cmd/kubeadm/app/util/config/common.go

```diff
@@ -234,6 +268,40 @@ func MigrateOldConfig(oldConfig []byte) ([]byte, error) {
 	return bytes.Join(newConfig, []byte(constants.YAMLDocumentSeparator)), nil
 }
 
+// ValidateConfig takes a byte slice containing a kubeadm configuration and performs conversion
+// to internal types and validation.
+func ValidateConfig(oldConfig []byte) error {
+	gvkmap, err := kubeadmutil.SplitYAMLDocuments(oldConfig)
+	if err != nil {
+		return err
+	}
+
+	gvks := []schema.GroupVersionKind{}
+	for gvk := range gvkmap {
+		gvks = append(gvks, gvk)
+	}
+
+	if err := verifyKnownGVKs(gvks); err != nil {
```

Comment:
When a user is using the v1beta2, it will just be shown as unknown.

For `kubeadm config migrate`, it will show the message that you can migrate v1beta2 to v1beta3 using kubeadm v1.22. I think the message helps when validating an old version of the kubeadm configuration. 

https://github.com/kubernetes/kubernetes/blob/ed1e45b1092a6d736895b6fd0764b2b01b258e04/cmd/kubeadm/app/util/config/common.go#L57-L58

Using `validateSupportedVersion` can show that message.

Reply:
i also think the message is valuable.

this is a bit problematic because the validateSupportedVersion function only validates a known GroupVersion. it does not know what Kinds existed in the version.

i need to think what is best here.

Reply:
updated to keep legacy versions in mind.